core/src/main/scala/org/bykn/bosatsu/Parser.scala

```diff
@@ -448,7 +448,11 @@ object Parser {
     (P.char('(') ~ ws) *> pa <* (ws ~ P.char(')'))
 
   val newline: P[Unit] = P.char('\n')
-  val toEOL: P0[Unit] = maybeSpace *> newline.orElse(P.end)
+  // The comment excluding the leading # up to the EOL or EOF
+  val termination: P0[Unit] = newline.orElse(P.end)
+  val commentToEOL: P[String] =
+    P.char('#') *> P.charsWhile0(_ != '\n') <* termination
```

Comment:
This only parses a single comment to EOL. We could have several. This isn't a problem for the code as it is now due to the way it's used but it's dangerous to leave it around since it isn't clear if it parses one or many lines.

Also it doesn't allow space then a comment which makes it rarely useful.